src/main/java/com/github/onsdigital/babbage/api/endpoint/content/Resource.java

```diff
@@ -23,13 +36,72 @@
+                contentResponse = ContentClient.getInstance().getResource(uri);
+            } catch (ContentReadException e) {
+                System.out.println("Get resource returned error response, status: " + e.getStatusCode());
+                ErrorHandler.renderErrorPage(404, response);
+                return;
+            }
+
             String contentDispositionHeader = "inline; ";
             contentDispositionHeader += contentResponse.getName() == null ? "" : "filename=\"" + contentResponse.getName() + "\"";
             response.setHeader("Content-Disposition", contentDispositionHeader);
-            new BabbageContentBasedBinaryResponse(contentResponse, contentResponse.getDataStream(), contentResponse.getMimeType()).apply(request, response);
+
+            try ( InputStream contentResponseBody = contentResponse.getDataStream() ) {
+
+                BufferedImage image = null;
+
+                if (width != null) {
+
+                    try {
+                        Integer w = Integer.parseInt(width, 10);
+
+                        if (uri.toLowerCase().endsWith(".png") || uri.toLowerCase().endsWith(".jpg") || uri.toLowerCase().endsWith(".jpeg")) {
+                            image = ImageIO.read(contentResponseBody);
+                            Integer clampedWidth = Math.max(1, Math.min(image.getWidth(), w));
+
+                            Double w2 = (double) clampedWidth;
+                            Double ratio = w2 / (double) image.getWidth();
+                            Double h = (double) image.getHeight() * ratio;
+
+                            Image newImage = image.getScaledInstance(clampedWidth, h.intValue(), Image.SCALE_SMOOTH);
+                            image = new BufferedImage(newImage.getWidth(null), newImage.getHeight(null), BufferedImage.TYPE_INT_RGB);
+                            image.getGraphics().drawImage(newImage, 0, 0, null);
+                        }
+
+                    } catch (IOException e) {
+                        // intentionally swallowing exception so we can return original image if resize fails
+                        e.printStackTrace();
+                    }
+
+                }
+
+
+                if (image != null) {
+
+
+                    try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+                        ImageIO.write(image, "png", os);
+                        try ( InputStream input = new ByteArrayInputStream(os.toByteArray())) {
+                            new BabbageContentBasedBinaryResponse(contentResponse, input, contentResponse.getMimeType()).apply(request, response);
+                            return;
+                        }
+
+                    }
+                }
+
+
+                new BabbageContentBasedBinaryResponse(contentResponse, contentResponseBody, contentResponse.getMimeType()).apply(request, response);
+            }
         } catch (Throwable t) {
             ErrorHandler.handle(request, response, t);
```

Comment:
I don't think this catch is required. Any exception thrown should already reach the error handler and have the same effect (+ logging)